dace/frontend/python/memlet_parser.py

```diff
@@ -0,0 +1,236 @@
+import ast
+import copy
+import re
+from collections import namedtuple
+from typing import Any, Dict, Union
+
+from dace import data, subsets
+from dace.frontend.python import astutils
+from dace.frontend.python.astutils import rname
+from dace.memlet import Memlet
+from dace.symbolic import pystr_to_symbolic
+
+MemletType = Union[ast.Call, ast.Attribute, ast.Subscript, ast.Name]
+MemletExpr = namedtuple('MemletExpr',
+                        ['name', 'accesses', 'wcr', 'wcr_identity', 'subset'])
+
+class DaceSyntaxError(Exception):
```

Comment:
not sure that it belongs here. Is there a reason why it's not in newast or a common.py?